packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { KeyboardEventHandler, useEffect, useState } from 'react'
+import {
+  Box,
+  List,
+  Paper,
+  Typography,
+  FormControl,
+  InputAdornment,
+  IconButton,
+} from '@mui/material'
+import { useTheme } from '@mui/material'
+import { ColorPartial } from '@mui/material/styles/createPalette'
+import { OutlinedInput } from '@mui/material'
+import { ArrowUpward } from '@mui/icons-material'
+import SynapseChatInteraction from './SynapseChatInteraction'
+import { SkeletonParagraph } from '../Skeleton'
+import { useCreateAgentSession } from 'src/synapse-queries/chat/useChat'
+import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+
+export type SynapseChatProps = {
+  initialMessage: string
+  agentId?: string // if provided, use this agent
+}
+
+export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
+  initialMessage,
+  agentId,
+}) => {
+  // Create a new Agent Session on mount, add ability to initialize this component with a starting message from the user.
+  const [agentSessionId, setAgentSessionId] = useState('')
+  const { mutate: createAgentSession } = useCreateAgentSession({
+    onSuccess: newAgentSession => {
+      setAgentSessionId(newAgentSession.sessionId)
+    },
+  })
+  const theme = useTheme()
+  const [messages, setMessages] = useState([initialMessage])
+  const [currentMessage, setCurrentMessage] = useState('')
+
+  useEffect(() => {
+    // on mount, create a new agent session!
+    if (createAgentSession) {
+      createAgentSession({
+        agentAccessLevel: AgentAccessLevel.PUBLICLY_ACCESSIBLE,
+        agentId: agentId,
+      })
+    }
+  }, [createAgentSession])
```

Comment:
```suggestion
  useEffect(() => {
    // on mount, create a new agent session!
    if (createAgentSession && initialMessage && !agentSessionId) {
      createAgentSession({
        agentAccessLevel: AgentAccessLevel.PUBLICLY_ACCESSIBLE,
        agentId: agentId,
      })
    }
  }, [createAgentSession, initialMessage, agentSessionId])
```

Just in case something changes later on and we render this before the user has typed a message, let's check that a message is present and that we haven't already started a session.

IMO this would be more useful when accessible from any page, not just a single chat page, so while that won't be how it works in this initial version, I feel like there's a decent chance we'll eventually end up there

Reply:
Like the agentSessionId check, let's not create a new agent session if we already have one!  Not sure about the initialMessage in this logic.  Whether we have an initial message or not, I think we should establish the agent session ID.